File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapConstructor.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapConstructors;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapConstructors.class)
public @interface WrapConstructor {
    MapEnum targetMap();

    Class<?>[] signature() default {};
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapConstructor.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapConstructors
import kotlin.reflect.KClass

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapConstructors::class
)
annotation class WrapConstructor(val targetMap: MapEnum, val signature: Array<KClass<*>> = [])